### ISSUE TITLE
Improve Documentation

### ISSUE THREAD

**Issue by pcn** [google/containerregistry#33]
The documentation includes examples of using the auth helpers for google cloud, aws, and azure, which together seems to indicate that it'd be expected for those to all work. However, aside from gcr the README doesn't clearly state whether the tools are expected to work with dockerhub.  I spent some time trying yesterday and it doesn't seem to work, but I don't know if that's as intended.

**Comment by mattmoor** [comment]
tl;dr You want `index.docker.io` and possibly `library/` before single-name images e.g. `ubuntu`.

Ack on docs.  This library is what sits at the heart of `bazelbuild/rules_docker`, so it has pretty broad compatibility (and that's where the most focus has gone into documentation).  But it notably doesn't special case DockerHub everywhere like the Docker CLI (you can get closer by disabling strictness [here](https://github.com/google/containerregistry/blob/master/client/docker_name_.py#L181)).

There are a number of registries listed explicitly [here](https://github.com/bazelbuild/rules_docker#container_pull-dockerhub), which we run integration tests against here:
 - DockerHub (fully supported)
 - Quay.io (supported here, but only "pull" in `rules_docker` because they lack v2.2 support)
 - Bintray / Artifactory (fully supported)
 - Gitlab (supported, but due to DNS issues there are caveats around the Python version)
 - GCR (fully supported)

We should also support at least ECR, which I haven't tested personally, but I know of several folks using `rules_docker` with them.  AzureCR I don't know.

**Comment by pcn** [comment]
It looks to me like this is now documented.
